core/p2p/core.go

```diff
@@ -87,8 +88,30 @@ func provide(c *dig.Container) {
 
 		ctx := context.Background()
 
+		identityExists := func(peerStorePath string) bool {
```

Comment:
Move this inline function out as this c.Provide() is already super long.

Reply:
Done.